src/Symfony/Component/Scheduler/Trigger/PeriodicalTrigger.php

```diff
@@ -0,0 +1,147 @@
+        $this->until = \is_string($until) ? new \DateTimeImmutable($until) : $until;
+
+        if (\is_int($interval)) {
+            self::ensurePositiveInteger($this->intervalInSeconds = $interval);
+
+            return;
+        }
+
+        if (is_string($interval) && ctype_digit($interval)) {
+            self::ensureIntervalSize($interval);
+            $this->intervalInSeconds = (int) $interval;
+
+            return;
+        }
+
+        try {
+            if (\is_string($interval) && 'P' === ($interval[0] ?? '')) {
+                $this->intervalInSeconds = self::calcInterval($this->from, $this->from->add(new \DateInterval($interval)));
+
+                return;
+            }
+
+            $i = $interval;
+            if (is_string($interval)) {
+                $this->description = $interval;
+                $i = \DateInterval::createFromDateString($interval);
+            } else {
+                $this->description = isset($interval->from_string) && $interval->from_string ? $interval->date_string : 'DateInterval';
+            }
+            if ($this->canBeConvertedToSeconds($i)) {
+                $this->intervalInSeconds = self::calcInterval($this->from, $this->from->add($i));
+            } else {
+                $this->period = new \DatePeriod($this->from, $i, $this->until);
+            }
+        } catch (\Exception $e) {
+            throw new InvalidArgumentException(sprintf('Invalid interval "%s": ', $interval instanceof \DateInterval ? 'instance of \DateInterval' : $interval).$e->getMessage(), 0, $e);
+        }
+    }
+
+    public function __toString(): string
+    {
+        return $this->description ?? sprintf('every %d seconds', $this->intervalInSeconds);
+    }
+
+    public function getNextRunDate(\DateTimeImmutable $run): ?\DateTimeImmutable
+    {
+        if ($this->intervalInSeconds) {
+            if ($this->from > $run) {
+                return $this->from;
+            }
+            if ($this->until <= $run) {
+                return null;
+            }
+
+            $delta = $run->format('U.u') - $this->from->format('U.u');
+            $recurrencesPassed = (int) ($delta / $this->intervalInSeconds);
+            $nextRunTimestamp = ($recurrencesPassed + 1) * $this->intervalInSeconds + $this->from->getTimestamp();
+            $nextRun = \DateTimeImmutable::createFromFormat('U.u', $nextRunTimestamp.$this->from->format('.u'), $this->from->getTimezone());
+
+            return $this->until > $nextRun ? $nextRun : null;
+        }
+
+        $iterator = $this->period->getIterator();
+        while ($run >= $next = $iterator->current()) {
+            $iterator->next();
+            if (!$iterator->valid()) {
+                return null;
+            }
+        }
+
+        return $next;
+    }
+
+    private function canBeConvertedToSeconds(\DateInterval $interval): bool
+    {
+        new \ReflectionProperty($interval, 'from_string');
```

Comment:
That's a hack but I didn't find any other way to access the `\DateInterval` properties. /cc @nicolas-grekas

Reply:
Can't you check for y, m, d to decide this function?

`return !$internal->y && !$interval->m && !$interval->d;`

Reply:
No, we cannot